### PR TITLE
Add webapp and hirni

### PR DESCRIPTION
**Pull request by bpoldrack**
- [x] NF: allow to specify github repos outside datalad organization
- [x] NF: allow to specify whether to use `nose` or `pytest` for an extension's tests
- [x] ENH: Add `datalad-webapp` to the to be tested extensions
- [x] ENH: Add `datalad-hirni``to the to be tested extensions

### CODE REVIEW

File: extensions.yaml [hunk 1]
```diff
@@ -28,5 +28,12 @@ extensions:
...
  datalad_crawler:
- # datalad_webapp,     # not developed ATM, not sure if of interest to keep track
+ datalad_webapp:
+  pypi: datalad-webapp
+  package: datalad_webapp
+ datalad_hirni:
+  apt_depends: [dcm2niix]
+  pypi: datalad-hirni
+  package: datalad_hirni
+  github: /psychoinformatics-de/datalad-hirni
```
Comment (yarikoptic) [inline extensions.yaml:38]: FWIW, I would have kept leading `/` out of this one here, and explicit in the urls like `https://github.com/{e. ...`

**Comment by yarikoptic** [comment]
bloody azure ... heh heh

File: extensions.yaml (hunk 1 above)
Comment (bpoldrack) [reply extensions.yaml:38]: Well, can do it that way, too. Leading `/` is just a slightly simpler thing to check for. Doesn't really matter to me.

**Comment by bpoldrack** [comment]
@yarikoptic : Changed to have that leading `/` and added singualrity as requirement for hirni tests. What's missing is to be able to specify a different testing framework, since webapp is written for pytest not nose. Looking into that the respective templating ...

**Comment by yarikoptic** [comment]
It's jinja so I guess you could add a conditional block and introduce a new configuration setting to extensions.yaml .

**Comment by bpoldrack** [comment]
Grmpf.

And: hirni test build didn't work, since it failed to run setup.py for `citeproc-py` (which doesn't happen locally for me, of course). Where does that even come from? `duecredit`?

**Comment by bpoldrack** [comment]
Ok, now `datalad-hirni` tests are failing, BUT: It's the actual tests! That's fine, I think, since the purpose of this repo and this PR is not to fix datalad or any extension, but to run those tests. They run. 
`datalad-neuroimaging` stalled trying to setup for more than an hour by now. Passed before, nothing changed in that setup - has to be azure or something.

So - I'd vote for merge. Whatever leads to hirni failure is to investigate, but again: The point is to have those tests running not passing.

**Pull request merged**